src/utils/decorators/tooltip-decorator/tooltip-decorator.js

```diff
@@ -163,198 +122,145 @@ const TooltipDecorator = (ComposedComponent) => {
+      const alignment = this.state.tooltipAlign || tooltipAlign || 'center';
+      const isTooltipOffScreen = window.innerWidth < shifts.vertical[alignment] + tooltipWidth;
 
-      const position = this.props.tooltipPosition || 'top';
-      const alignment = this.props.tooltipAlign || 'center';
+      if (alignment === 'right' || position === 'left' || !isTooltipOffScreen) return;
 
-      if (position === 'top' || position === 'bottom') {
-        const horizontalPosition = shifts[`vertical${startCase(alignment)}`];
-        if (window.innerWidth < (horizontalPosition + tooltipWidth)) {
-          this.setState({ tooltipAlign: 'right' });
-        }
+      if (position === 'right') {
+        this.setState({ tooltipPosition: 'top', tooltipAlign: 'right' });
+      } else {
+        this.setState({ tooltipAlign: 'right' });
       }
     }
 
-    /**
-     * Positions tooltip relative to target
-     *
-     * @method positionTooltip
-     * @param {Object} tooltip
-     * @param {Object} target
-     * @return {Void}
-     */
     positionTooltip = () => {
-      if (this.isVisible()) {
-        const tooltip = this.getTooltip(),
-            target = this.getTarget();
-        if (!tooltip || !target) {
-          // Can't find the tooltip or target so hide
-          if (this.state.isVisible) this.setState({ isVisible: false });
-          return;
-        }
-
-        const alignment = this.state.tooltipAlign || this.props.tooltipAlign || 'center',
-            position = this.props.tooltipPosition || 'top',
-            shifts = this.calculatePosition(tooltip, target);
-
-        switch (position) {
-          case 'top':
-            styleElement(tooltip, 'top', append(shifts.verticalY, 'px'));
-            styleElement(tooltip, 'right', 'auto');
-            styleElement(tooltip, 'bottom', 'auto');
-            styleElement(tooltip, 'left', append(shifts[`vertical${startCase(alignment)}`], 'px'));
-            break;
-
-          case 'bottom':
-            styleElement(tooltip, 'top', append(shifts.verticalBottomY, 'px'));
-            styleElement(tooltip, 'bottom', 'auto');
-            styleElement(tooltip, 'left', append(shifts[`vertical${startCase(alignment)}`], 'px'));
-            break;
-
-          case 'left':
-            styleElement(tooltip, 'top', append(shifts[`side${startCase(alignment)}`], 'px'));
-            styleElement(tooltip, 'bottom', 'auto');
-            styleElement(tooltip, 'left', append(shifts[`${position}Horizontal`], 'px'));
-            break;
-
-          case 'right':
-          default:
-            styleElement(tooltip, 'top', append(shifts[`side${startCase(alignment)}`], 'px'));
-            styleElement(tooltip, 'bottom', 'auto');
-            styleElement(tooltip, 'left', append(shifts[`${position}Horizontal`], 'px'));
-        }
+      if (!this.isVisible()) {
+        return;
+      }
+
+      const tooltip = this.getTooltip(),
+          target = this.getTarget();
```

Comment:
Just little one, would you like to create both const the same way?
I mean:
```
const tooltip = this.getTooltip();
const target = this.getTarget();
```
I think this is more readable